ci/builder/Dockerfile

```diff
@@ -293,17 +293,17 @@ RUN mkdir rust \
     && gpg --verify rust.asc rust.tar.gz \
     && tar -xzf rust.tar.gz -C /usr/local/lib/rustlib/ --strip-components=4 \
     && rm -rf rust.asc rust.tar.gz rust \
-    && cargo install --root /usr/local --version "=0.6.6" --locked cargo-about \
-    && cargo install --root /usr/local --version "=2.0.2" --locked cargo-deb \
-    && cargo install --root /usr/local --version "=0.14.21" --locked cargo-deny \
+    && cargo install --root /usr/local --version "=0.7.1" --locked cargo-about \
+    && cargo install --root /usr/local --version "=3.2.0" --locked cargo-deb \
+    && cargo install --root /usr/local --version "=0.18.3" --locked cargo-deny \
     && cargo install --root /usr/local --version "=0.1.0" --locked cargo-deplint \
-    && cargo install --root /usr/local --version ="0.9.28" --locked cargo-hakari \
+    && cargo install --root /usr/local --version "=0.9.36" --locked cargo-hakari \
     && cargo install --root /usr/local --version "=0.9.72" --locked cargo-nextest \
```

Comment:
nextest will already be updated in https://github.com/MaterializeInc/materialize/pull/32949